contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -1719,7 +1720,16 @@ PrepareRowDescription(TupleDesc typeinfo, List *targetlist, int16 *formats,
 
 				/* fetch the relation name, schema name */
 				relMetaDataInfo->partName[0] = RelationGetRelationName(rel);
-				relMetaDataInfo->partName[1] = get_namespace_name(RelationGetNamespace(rel));
+				physical_schema_name = get_namespace_name(RelationGetNamespace(rel));
+
+				/* logical schema name should be sent as part of response as far as Babelfish is concerned. */
+				if (pltsql_plugin_handler_ptr && 
+					pltsql_plugin_handler_ptr->pltsql_get_logical_schema_name)
+					/* Its very unlikely that logical schema name could not be found for any other cases. */
+					relMetaDataInfo->partName[1] = pltsql_plugin_handler_ptr->pltsql_get_logical_schema_name(physical_schema_name);
+
+				if (physical_schema_name)
```

Comment:
Is it safe to delete this? it is possible that partName[1] is pointing to this schema name (for sys and information schema)

Reply:
Yes, it shouldn't get freed. Updated this in next revision.